src/msg/async/rdma/RDMAConnectedSocketImpl.cc

```diff
@@ -517,11 +551,16 @@ void RDMAConnectedSocketImpl::cleanup() {
   if (con_handler && tcp_fd >= 0) {
     (static_cast<C_handle_connection*>(con_handler))->close();
     worker->center.submit_to(worker->center.get_id(), [this]() {
-      worker->center.delete_file_event(tcp_fd, EVENT_READABLE);
+      worker->center.delete_file_event(tcp_fd, EVENT_READABLE | EVENT_WRITABLE);
     }, false);
     delete con_handler;
     con_handler = nullptr;
   }
+  if (con_init_handler) {
+    (static_cast<C_handle_connection_init*>(con_init_handler))->close();
+    delete con_init_handler;
+    con_init_handler = nullptr;
```

Comment:
no need to check for null before `delete con_init_handler`. why would you need to reset it as the only caller of this function is `~RDMAConnectedSocketImpl()`. so nobody will get a chance to dereference it after that.

Reply:
for the same reason, no need to `close()` `con_init_handler` for resetting "active".

Reply:
> no need to check for null before `delete con_init_handler`. why would you need to reset it as the only caller of this function is `~RDMAConnectedSocketImpl()`. so nobody will get a chance to dereference it after that.

`cleanup()`  maybe be called by ` RDMAConnectedSocketImpl::read()`。